src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -72,6 +72,26 @@ public sealed class CompletionItem : IComparable<CompletionItem>
             this.Rules = rules ?? CompletionItemRules.Default;
         }
 
+#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads.
```

Comment:
No.  it should not be an issue.  It doesn't apply when one of the overloads is being deprecated.  The other overload stays, but loses all optional parameters (so that we retain binary compat).

This overload is now introduced and is what all source calls should now call into.